docs/source/1.0/spec/aws/aws-cloudformation.rst

```diff
@@ -0,0 +1,780 @@
+resource provider`_ for use within CloudFormation. Smithy's modeled
+:ref:`resources <resource>`, utilizing the traits below, can generate these
+schemas. Automatically generating schemas from a service's API lowers the
+effort needed to generate and maintain them, reduces the potential for errors
+in the translation, and provides a more complete depiction of a resource in its
+schema. These schemas can be utilized by the `CloudFormation Command Line
+Interface`_ to build, register, and deploy `resource providers`_.
+
+.. contents:: Table of contents
+    :depth: 3
+    :local:
+    :backlinks: none
+
+
+.. _aws.cloudformation#resource-trait:
+
+-------------------------------------
+``aws.cloudformation#resource`` trait
+-------------------------------------
+
+Summary
+    Indicates that a Smithy resource is a CloudFormation resource.
+Trait selector
+    ``resource``
+Value type
+    ``structure``
+
+The ``aws.cloudformation#resource`` trait is a structure that supports the
+following members:
+
+.. list-table::
+    :header-rows: 1
+    :widths:  10 20 70
+
+    * - Property
+      - Type
+      - Description
+    * - name
+      - ``string``
+      - Provides a custom CloudFormation resource name. This defaults to the
+        shape name component of the ``resource`` shape's :ref:`shape
+        ID <shape-id>`.
+    * - additionalSchemas
+      - ``list<shapeId>``
+      - A list of additional :ref:`shape IDs <shape-id>` of structures that
+        will have their properties added to the CloudFormation resource.
+        Members of these structures with the same names MUST resolve to the
+        same target. See :ref:`aws-cloudformation-property-deriviation` for
+        more information.
+
+The following example defines a simple resource that is also a CloudFormation
+resource:
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        namespace smithy.example
+
+        use aws.cloudformation#resource
+
+        @resource
+        resource Foo {
+            identifiers: {
+                fooId: String,
+            },
+        }
+
+
+The following example provides a ``name`` value and one structure shape in the
+``additionalSchemas`` list.
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        namespace smithy.example
+
+        use aws.cloudformation#resource
+
+        @resource(
+            name: "Foo",
+            additionalSchemas: [AdditionalFooProperties])
+        resource FooResource {
+            identifiers: {
+                fooId: String,
+            },
+        }
+
+        structure AdditionalFooProperties {
+            barProperty: String,
+        }
+
+
+.. _aws-cloudformation-property-deriviation:
+
+Resource properties
+===================
+
+Smithy will automatically derive `property`__ information for resources with the
+``@aws.cloudformation#resource`` trait applied.
+
+A resource's properties include the resource's identifiers as well as the top
```

Comment:
Are these Smithy resource identifiers? If so, can you link to that part of the spec?